Foundation/Process.swift

```diff
@@ -164,13 +164,40 @@ open class Process: NSObject {
     
     }
     
+    @available(*, deprecated)
```

Comment:
These are actually not marked as deprecated yet in the SDK.

Reply:
The new implementation looks reasonable but we can leave the deprecation annotation off for now.

Reply:
It is marked as deprecated in the documentation: https://developer.apple.com/documentation/foundation/process/1413110-currentdirectorypath

Maybe it is just an error.

Reply:
If the path API is not deprecated I would prefer to do the inverted implementation (just add the new URL API as an overlay of the path counterparts).

Reply:
The documentation actually got ahead of us here. It is "soft deprecated", indicating that will eventually be deprecated.

It may seem like splitting hairs but I think we should remain consistent with the annotations in the SDK. Once the SDK marks it deprecated (with a `API_DEPRECATED` macro, e.g.) then we can add the attribute here.

If you want to invert the implementation, that's fine with me. That's actually how we do it in the ObjC Foundation too.